### PR TITLE
Run for job for all currencies, and check in bzip list + checksum

### PR DESCRIPTION
**Pull request by nvonpentz**
For https://github.com/brave-intl/compliance-ops/blob/main/record-keeping.md

### CODE REVIEW

**Review by mrose17 — CHANGES_REQUESTED** [review]
Overall, this looks good. There's one thing though: when the checksum is calculated, it needs to be compared to the checksum that is on the website to see if they match. If they don't then it's not all that useful. 

perhaps download and save https://sanctionslist.ofac.treas.gov/Home/SdnList and then parse the file (painful, I know) in order to verify the file signature. it is annoying that they don't include the signature information as a file for download.

File: .github/workflows/generate-lists.yml [hunk 1]
```diff
@@ -22,20 +21,38 @@ jobs:
       uses: actions/setup-python@82c7e631bb3cdc910f68e0081d67478d79c6982d # v5
       with:
         python-version: 3.11
+    - name: Install dependencies
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y wget gnupg
+        wget -q -O - https://dl.google.com/linux/linux_signing_key.pub | sudo apt-key add -
+        sudo sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google-chrome.list'
```
[the remark this answers is not in the document]
Comment (thypon) [reply .github/workflows/generate-lists.yml:29]: false positive

File: generate-address-list.py [hunk 2]
```diff
@@ -74,10 +87,31 @@ def write_addresses_json(addresses, asset, outpath):
...
+    # First, check if the checksum matches the one on the website
+    scraper = OfacWebsiteScraper()
+    try:
+        sha256_checksum_from_site = scraper.get_sha256_checksum()
+    finally:
+        scraper.close()
+
+    sha256_checksum_computed = compute_sha256(SDN_ADVANCED_FILE_PATH)
+    if sha256_checksum_computed != sha256_checksum_from_site:
+        raise "Checksums do not match."
```
Comment (kdenhartog) [inline generate-address-list.py:113]: probably worth alerting with the bot on this scenario as well

File: ofac_scraper.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        self.open_website("https://sanctionslist.ofac.treas.gov/Home/SdnList")
+
+        # Wait until the 'File Signatures' button with the known ID is present
+        self.wait_for_element(By.ID, "accordion__heading-raa-1")
+
+        # Click the 'File Signatures' button with the known ID
+        self.click_element(By.ID, "accordion__heading-raa-1")
+
+        # Wait for the checksums panel to be visible
+        self.wait_for_element(By.ID, "accordion__panel-raa-1")
```
Comment (kdenhartog) [inline ofac_scraper.py:49]: We should likely log and error if these elements aren't found in case the page is changed in some way

**Review by kdenhartog — APPROVED** [review]
few nits to detect errors a bit more, but seems fine to me

File: generate-address-list.py (hunk 2 above)
Comment (nvonpentz) [reply generate-address-list.py:113]: Should do so on an uncaught error

File: .github/workflows/generate-lists.yml [hunk 4]
```diff
@@ -22,24 +21,61 @@ jobs:
...
         github_token: ${{ secrets.GITHUB_TOKEN }}
         branch: lists
+
+  failure-notification:
+    runs-on: ubuntu-latest
+    needs: generate-lists
+    if: failure()
+    steps:
+      - name: Notify Slack of failure
+        uses: 8398a7/action-slack@v3
```
[the remark this answers is not in the document]
Comment (thypon) [reply .github/workflows/generate-lists.yml:70]: can you fix this one? @nvonpentz

[the remark this answers is not in the document]
Comment (nvonpentz) [reply .github/workflows/generate-lists.yml]: Updated to
```
uses: 8398a7/action-slack@28ba43ae48961b90635b50953d216767a6bea486 # https://github.com/8398a7/action-slack/releases/tag/v3.16.2
```

**Review by mrose17 — APPROVED** [review]
Good to go!

**Pull request merged**